pkg/analysis/optionalfields/analyzer.go

```diff
@@ -300,7 +314,7 @@ func getStructZeroValue(pass *analysis.Pass, structType *ast.StructType) string
 	for _, field := range structType.Fields.List {
 		fieldTagInfo := jsonTagInfo.FieldTags(field)
 
-		if fieldTagInfo.OmitEmpty {
+		if fieldTagInfo.OmitEmpty || fieldTagInfo.OmitZero {
```

Comment:
I _think_ we can only rely on `omitzero` when the field isn't a pointer, can we check that here?